panacea/grpc_client.go

```diff
@@ -63,24 +78,142 @@ func (c *GrpcClient) GetPubKey(panaceaAddr string) ([]byte, error) {
+	client := authtypes.NewQueryClient(c.conn)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	response, err := client.Account(ctx, &authtypes.QueryAccountRequest{Address: panaceaAddr})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get account info via grpc: %w", err)
+	}
+
+	var acc authtypes.AccountI
+	if err := c.interfaceRegistry.UnpackAny(response.GetAccount(), &acc); err != nil {
+		return nil, fmt.Errorf("failed to unpack account info: %w", err)
+	}
+	return acc, nil
 }
 
 // GetDeal gets deal info from blockchain
-func (c *GrpcClient) GetDeal(id string) (markettypes.Deal, error) {
+func (c *GrpcClient) GetDeal(id string) (datadealtypes.Deal, error) {
 	dealId, err := strconv.ParseUint(id, 10, 64)
 	if err != nil {
-		return markettypes.Deal{}, fmt.Errorf("failed to parse deal id: %w", err)
+		return datadealtypes.Deal{}, fmt.Errorf("failed to parse deal id: %w", err)
 	}
 
-	client := markettypes.NewQueryClient(c.conn)
+	client := datadealtypes.NewQueryClient(c.conn)
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
 	defer cancel()
 
-	response, err := client.Deal(ctx, &markettypes.QueryDealRequest{DealId: dealId})
+	response, err := client.Deal(ctx, &datadealtypes.QueryDealRequest{DealId: dealId})
 	if err != nil {
-		return markettypes.Deal{}, fmt.Errorf("failed to get deal info: %w", err)
+		return datadealtypes.Deal{}, fmt.Errorf("failed to get deal info: %w", err)
 	}
 
 	return *response.GetDeal(), nil
 }
+
+// RegisterDataValidator registers data validator on blockchain
+func (c *GrpcClient) RegisterDataValidator(address, endpoint string, validatorAcc *ValidatorAccount) error {
+	interfaceRegistry := makeInterfaceRegistry()
+	marshaler := codec.NewProtoCodec(interfaceRegistry)
+	txConfig := tx.NewTxConfig(marshaler, []signing.SignMode{signing.SignMode_SIGN_MODE_DIRECT})
+	txBuilder := txConfig.NewTxBuilder()
+
+	dataValidator := &datapooltypes.DataValidator{
+		Address:  address,
+		Endpoint: endpoint,
+	}
+
+	msgRegisterDataValidator := datapooltypes.NewMsgRegisterDataValidator(dataValidator)
+
+	err := txBuilder.SetMsgs(msgRegisterDataValidator)
+	if err != nil {
+		return err
+	}
+
+	privKey := secp256k1.PrivKey{
+		Key: validatorAcc.secp256k1PrivKey.Bytes(),
+	}
+
+	account, err := c.GetAccount(address)
+	if err != nil {
+		return err
+	}
+
+	sequence := account.GetSequence()
+
+	//TODO: Fee will be set in Config.toml in near future, now just hard-coded.
+	fees := cosmostype.NewCoins(cosmostype.NewInt64Coin("umed", 1000000))
+	txBuilder.SetFeeAmount(fees)
+	txBuilder.SetGasLimit(200000)
+
+	var sigsV2 []signing.SignatureV2
+	sigV2 := signing.SignatureV2{
+		PubKey: privKey.PubKey(),
+		Data: &signing.SingleSignatureData{
+			SignMode:  signing.SignMode_SIGN_MODE_DIRECT,
+			Signature: nil,
+		},
+		Sequence: sequence,
+	}
+	sigsV2 = append(sigsV2, sigV2)
+
+	err = txBuilder.SetSignatures(sigsV2...)
```

Comment:
You could also use it like this.(maybe?)
```suggestion
	sigV2 := signing.SignatureV2{
		PubKey: privKey.PubKey(),
		Data: &signing.SingleSignatureData{
			SignMode:  signing.SignMode_SIGN_MODE_DIRECT,
			Signature: nil,
		},
		Sequence: sequence,
	}

	err = txBuilder.SetSignatures(sigV2)
```